handler/schedule_test.go

```diff
@@ -146,12 +146,161 @@ func TestScheduleHandler_mergeResourcesWithDefaults(t *testing.T) {
 			cfg.Config.GlobalMemoryResourceLimit = tt.globalMemoryResourceLimit
 			cfg.Config.GlobalMemoryResourceRequest = tt.globalMemoryResourceRequest
 			require.NoError(t, cfg.Config.ValidateSyntax())
-			s := ScheduleHandler{schedule: &v1alpha1.Schedule{Spec: v1alpha1.ScheduleSpec{
-				ResourceRequirementsTemplate: tt.template,
+			schedule := ScheduleHandler{schedule: &v1alpha1.Schedule{Spec: v1alpha1.ScheduleSpec{
+				ResourceRequirementsTemplate: tt.givenScheduleTemplate,
 			}}}
-			spec := &v1alpha1.RunnableSpec{Resources: tt.resources}
-			s.mergeResourcesWithDefaults(spec)
-			assert.Equal(t, tt.expected, spec.Resources)
+			res := &v1alpha1.RunnableSpec{
+				Resources: tt.givenResourceTemplate,
+			}
+			schedule.mergeResourcesWithDefaults(res)
+			assert.Equal(t, tt.expectedTemplate, res.Resources)
+		})
+	}
+}
+
+func TestScheduleHandler_mergeBackendWithDefaults(t *testing.T) {
+	tests := []struct {
+		name                 string
+		globalS3Bucket       string
+		givenScheduleBackend v1alpha1.Backend
+		givenResourceBackend v1alpha1.Backend
+		expectedBackend      v1alpha1.Backend
+	}{
+		{
+			name: "Given_NoGlobalDefaults_And_NoScheduleDefaults_When_Spec_Then_UseSpec",
+			givenResourceBackend: v1alpha1.Backend{
```

Comment:
I suggest to create a small helper function that improves readability
```suggestion
			givenResourceBackend: createBackend("https://resource-url", "resource-bucket")
```

Reply:
I've applied the same for the other test and the `ResourceList` resource.